provider.go

```diff
@@ -135,6 +126,53 @@ func (g *InstanceGroup) Init(ctx context.Context, log hclog.Logger, settings pro
 	}, nil
 }
 
+func (g *InstanceGroup) getProviderClient(ctx context.Context) (*gophercloud.ProviderClient, gophercloud.EndpointOpts, error) {
+	var endpointOps gophercloud.EndpointOpts
+	var authOptions gophercloud.AuthOptions
+	var providerClient *gophercloud.ProviderClient
+
+	if g.AuthFromEnv {
+		g.log.Debug("Using env vars for auth")
+
+		var err error
+		endpointOps = gophercloud.EndpointOpts{Region: os.Getenv("OS_REGION_NAME")}
+		authOptions, err = openstack.AuthOptionsFromEnv()
```

Comment:
It not the best, as it more identity v2 compatible than v3. In one of utilities i had to reinvent the wheel to have proper operation, like with python lib.

Reply:
What would be the v3 version of it? 🤔 Currently this is the documented way shown in the gophercloud [README.md](https://github.com/gophercloud/gophercloud/blob/1678b7f36d39d983b4fec5995373e66a5ec3d738/README.md?plain=1#L110).

Reply:
Whatever they say, it's incompatible with openrc generated for v3. And even more, it breaks python agents, when they see OS_DOMAIN for example.

Here few parts of code, which do a proper support for v3:
https://gist.github.com/vooon/911c13af31c4aff53477536d8d6afe65

Reply:
Well I think it depends on what environment variables you configure. If you're supposed to talk to a v3 API, it could be very well be that configuring `OS_DOMAIN` breaks the tooling, because it's not supported. Which is definitely wrong, but I hesitate to add >100 lines of code just to catch those cases. How about you open a bug for gophercloud? Or would you prefer me to add those code lines from you Gist?

Reply:
I think GC would not break existing calls, as v2-rc time passed... Easier if you add similar code here. Not necessary the same way.
I've used interface because i reuse that common code in a lot of places, and it's helpful to embed into Kong cli...

Reply:
@baurmatt added. But please do not midlessly copy-paste it. OpenTelemetry for example is useless here...

Reply:
I've refactored the code to a proper openstackclient package, similar to what Gitlab does in their aws/gcp/azure plugins, which gives a much cleaner interface.

About the usage of `openstack.AuthOptionsFromEnv()`: I'm still not convinced that it's worth to reinvent the wheel. I've successfully tested my implementation and can confirm that it works.

Reply:
Thanks for the complete Gist! I've looked further into it and realized that it basically depends on kongs implementation of golang struct tag resolving (e.g. `env` and (especially) `dom`) to unfold its full effect. Now I'm even more hesitant to implement something like this. Is there anything in particular you would like to catch when using environment variables?

Reply:
github.com/caarlos0/env/v10 works (almost) the same.

Reply:
I think it's not necessary to support `xor` on env parsing, you may do the check later.
It just handy to error-exit early during cli opt validation.

Reply:
Alright, implemented something similar. Hope that's fine! :)